runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -84,6 +84,27 @@ computeServerMemoryState(TR::CompilationInfo *compInfo)
    return JITServer::ServerMemoryState::NORMAL;
    }
 
+/**
+ * @brief Helper method executed at the end of a compilation
+ * to determine whether the server is approaching the max. number of active threads.
+ */
+JITServer::ServerActiveThreadState
+computeServerActiveThreadState(TR::CompilationInfo *compInfo)
+   {
+   uint64_t highActiveThreadThreshold = 50;
+   uint64_t veryHighActiveThreadThreshold = 60;
+
+   uint64_t numOfActiveThreads = compInfo->getNumCompThreadsActive();
```

Comment:
Here we need a comment saying that typically `getNumCompThreadsActive()` needs to be protected the compilationQueueMonitor, but here we allow some small imprecision because we implement a heuristic.